template/sample/check/CheckTesterProcessor.gen.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Istio Authors.
```

Comment:
The validation of the config (just the adapter config) will be done by the Validation fn passed in the BuilderInfo passed to the mixer via the inventory of all adapters https://github.com/istio/mixer/blob/master/pkg/adapter/builderInfo.go#L24

For the validation of each template specific 'Type' (inferred types), the Configure* (example ConfigureMyMetric etc.) methods will be returning error. Adapter is suppose to validation there if they do not like the Types passed to them.

So, at least in my mind we do not have any open validation story to investigate.